lib/ansible/modules/cloud/azure/azure_rm_disk.py

```diff
@@ -0,0 +1,352 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Yuwei Zhou, <yuwzho@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'certified'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_disk
+version_added: "2.5"
+short_description: Manage Azure disk.
+description:
+    - Create, update or delete a disk. Attach or dettache the disk within a given virtual machine.
+options:
+    resource_group:
+        description:
+            - Name of resource group.
+        required: true
+    name:
+        description:
+            - Name of the disk.
+        required: true
+    location:
+        description:
+            - Location of the disk.
+        required: false
+    size_gb:
+        description:
+            - Size of the disk.
+        required: true
+    sku:
+        description:
+            - Underlying storage sku.
+        required: false
+        default: Premium_LRS
+        choices:
+            - Premium_LRS
+            - Standard_LRS
+    state:
+        description:
+            - Assert the state of the disk. Use 'present' to create or update a disk and
+              'absent' to delete a disk.
+        required: false
+        default: present
+        choices:
+            - absent
+            - present
+    managed_by:
+        description:
+            - Name of an existing virtual machine with which the disk is or will be associated.
+        required: false
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Yuwei Zhou (@yuwzho)"
+
+'''
+
+EXAMPLES = '''
+    - name: Create a disk and attach it to vm
+      azure_rm_disk:
+        resource_group: Testing
+        name: foobar
+        size_gb: 32
+        sku: Standard_LRS
+        managed_by: testvm
+
+    - name: Detach a disk
+      azure_rm_disk:
+        resource_group: Testing
+        name: foobar
+        size_gb: 32
+        sku: Standard_LRS
+    
+    - name: Delete a disk
+      azure_rm_disk:
+        state: absent
+        resource_group: Testing
+        name: foobar
+        size_gb: 32

```

Comment:
are `size_gb` and `sku` necessary for deletion?

Reply:
i think the only parameters to delete resource are resource group and resource name, so any other parameters should be optional.